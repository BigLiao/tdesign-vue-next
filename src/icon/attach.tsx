import Vue from 'vue';
import { prefix } from '../config';
import CLASSNAMES from '../utils/classnames';

const name = `${prefix}-icon-attach`;

export default Vue.extend({
  name,

  props: {
    // xs/small/middle/large/xl/18px/2em
    size: {
      type: String,
    },
  },

  computed: {
    classes(): Array<string|object> {
      return [
        `${prefix}-icon`,
        name,
        CLASSNAMES.SIZE[this.size],
      ];
    },
    iconStyle(): object {
      if (['xs', 'small', 'middle', 'large', 'xl'].includes(this.size)) return {};
      return {
        'font-size': this.size,
      };
    },
  },

  render() {
    return <svg class={this.classes} style={this.iconStyle}  viewBox="0 0 16 16" version="1.1" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M8.17636011,2.58820454 C8.95639503,1.83397426 10.1970826,1.84440701 10.9643237,2.61164811 C11.6907332,3.3380576 11.7289653,4.49204572 11.07902,5.2634635 L10.9643237,5.38835189 L6.46597753,9.8866981 L5.75887075,9.17959132 L10.257217,4.68124511 C10.6334582,4.30500382 10.6334582,3.69499618 10.257217,3.31875489 C9.90519905,2.96673699 9.35266853,2.93524024 8.96480093,3.22733535 L8.87147583,3.3071024 L3.55754515,8.44668263 C2.79326446,9.23499116 2.81274287,10.4936134 3.6010514,11.2578941 C4.38712348,12.0200065 5.6159989,12.0489078 6.43562317,11.3511907 L6.56151232,11.2349874 L12.5530364,5.24346336 L13.2601432,5.95057015 L7.26861911,11.9420942 C6.06742971,13.1432836 4.12460407,13.1583163 2.90497114,11.9758581 C1.76753522,10.8730915 1.69505694,9.08557958 2.70738827,7.89608211 L2.83958117,7.75060238 L2.90785324,7.68242191 L8.17636011,2.58820454 Z"/></svg>;
  },

});
