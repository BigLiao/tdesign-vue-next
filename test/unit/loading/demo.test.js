import { mount } from '@vue/test-utils';
import base from '@/examples/loading/demos/base.vue';
import delay from '@/examples/loading/demos/delay.vue';
import fullscreen from '@/examples/loading/demos/fullscreen.vue';
import iconText from '@/examples/loading/demos/icon-text.vue';
import attach from '@/examples/loading/demos/attach.vue';
import service from '@/examples/loading/demos/service.vue';
import size from '@/examples/loading/demos/size.vue';
import text from '@/examples/loading/demos/text.vue';
import wrap from '@/examples/loading/demos/wrap.vue';

// unit test for component in examples.
describe('Loading', () => {
  it('base demo works fine', () => {
    const wrapper = mount(base);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('delay demo works fine', () => {
    const wrapper = mount(delay);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('fullscreen demo works fine', () => {
    const wrapper = mount(fullscreen);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('iconText demo works fine', () => {
    const wrapper = mount(iconText);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('attach demo works fine', () => {
    const wrapper = mount(attach);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('service demo works fine', () => {
    const wrapper = mount(service);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('size demo works fine', () => {
    const wrapper = mount(size);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('text demo works fine', () => {
    const wrapper = mount(text);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('wrap demo works fine', () => {
    const wrapper = mount(wrap);
    expect(wrapper.element).toMatchSnapshot();
  });
});
