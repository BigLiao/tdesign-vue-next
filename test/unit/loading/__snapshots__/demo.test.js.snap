// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading base demo works fine 1`] = `
<span
  class="t-loading t-size-m"
>
  <svg
    class="t-icon t-icon-loading"
  >
    <use
      xlink:href="#t-icon-loading"
    />
  </svg>
</span>
`;

exports[`Loading delay demo works fine 1`] = `
<!--function (el) {
      var obj;

      var args = [], len = arguments.length - 1;
      while ( len-- &gt; 0 ) args[ len ] = arguments[ len + 1 ];
      if (shouldNotBeStubbed(el, stubs, modifiedComponents)) {
        return originalCreateElement.apply(void 0, [ el ].concat( args ))
      }

      if (isConstructor(el) || isComponentOptions(el)) {
        if (stubAllComponents) {
          var stub = createStubFromComponent(el, el.name || 'anonymous', _Vue);
          return originalCreateElement.apply(void 0, [ stub ].concat( args ))
        }
        var Constructor = shouldExtend(el, _Vue) ? extend(el, _Vue) : el;

        return originalCreateElement.apply(void 0, [ Constructor ].concat( args ))
      }

      if (typeof el === 'string') {
        var original = resolveComponent(el, originalComponents);

        if (!original) {
          return originalCreateElement.apply(void 0, [ el ].concat( args ))
        }

        if (isDynamicComponent(original)) {
          return originalCreateElement.apply(void 0, [ el ].concat( args ))
        }

        var stub$1 = createStubIfNeeded(stubAllComponents, original, _Vue, el);

        if (stub$1) {
          Object.assign(vm.$options.components, ( obj = {}, obj[el] = stub$1, obj));
          modifiedComponents.add(el);
        }
      }

      return originalCreateElement.apply(void 0, [ el ].concat( args ))
    }-->
`;

exports[`Loading fullscreen demo works fine 1`] = `
<!--function (el) {
      var obj;

      var args = [], len = arguments.length - 1;
      while ( len-- &gt; 0 ) args[ len ] = arguments[ len + 1 ];
      if (shouldNotBeStubbed(el, stubs, modifiedComponents)) {
        return originalCreateElement.apply(void 0, [ el ].concat( args ))
      }

      if (isConstructor(el) || isComponentOptions(el)) {
        if (stubAllComponents) {
          var stub = createStubFromComponent(el, el.name || 'anonymous', _Vue);
          return originalCreateElement.apply(void 0, [ stub ].concat( args ))
        }
        var Constructor = shouldExtend(el, _Vue) ? extend(el, _Vue) : el;

        return originalCreateElement.apply(void 0, [ Constructor ].concat( args ))
      }

      if (typeof el === 'string') {
        var original = resolveComponent(el, originalComponents);

        if (!original) {
          return originalCreateElement.apply(void 0, [ el ].concat( args ))
        }

        if (isDynamicComponent(original)) {
          return originalCreateElement.apply(void 0, [ el ].concat( args ))
        }

        var stub$1 = createStubIfNeeded(stubAllComponents, original, _Vue, el);

        if (stub$1) {
          Object.assign(vm.$options.components, ( obj = {}, obj[el] = stub$1, obj));
          modifiedComponents.add(el);
        }
      }

      return originalCreateElement.apply(void 0, [ el ].concat( args ))
    }-->
`;

exports[`Loading indicatorFunc demo works fine 1`] = `
<!--function (el) {
      var obj;

      var args = [], len = arguments.length - 1;
      while ( len-- &gt; 0 ) args[ len ] = arguments[ len + 1 ];
      if (shouldNotBeStubbed(el, stubs, modifiedComponents)) {
        return originalCreateElement.apply(void 0, [ el ].concat( args ))
      }

      if (isConstructor(el) || isComponentOptions(el)) {
        if (stubAllComponents) {
          var stub = createStubFromComponent(el, el.name || 'anonymous', _Vue);
          return originalCreateElement.apply(void 0, [ stub ].concat( args ))
        }
        var Constructor = shouldExtend(el, _Vue) ? extend(el, _Vue) : el;

        return originalCreateElement.apply(void 0, [ Constructor ].concat( args ))
      }

      if (typeof el === 'string') {
        var original = resolveComponent(el, originalComponents);

        if (!original) {
          return originalCreateElement.apply(void 0, [ el ].concat( args ))
        }

        if (isDynamicComponent(original)) {
          return originalCreateElement.apply(void 0, [ el ].concat( args ))
        }

        var stub$1 = createStubIfNeeded(stubAllComponents, original, _Vue, el);

        if (stub$1) {
          Object.assign(vm.$options.components, ( obj = {}, obj[el] = stub$1, obj));
          modifiedComponents.add(el);
        }
      }

      return originalCreateElement.apply(void 0, [ el ].concat( args ))
    }-->
`;

exports[`Loading indicatorSlot demo works fine 1`] = `
<div>
  <span
    class="t-loading t-size-m"
  >
    <i
      class="t-icon t-icon-loading"
    />
  </span>
</div>
`;

exports[`Loading preventScrollThrough demo works fine 1`] = `
<span
  class="t-loading t-size-m"
>
  <svg
    class="t-icon t-icon-loading"
  >
    <use
      xlink:href="#t-icon-loading"
    />
  </svg>
</span>
`;

exports[`Loading service demo works fine 1`] = `
<!--function (el) {
      var obj;

      var args = [], len = arguments.length - 1;
      while ( len-- &gt; 0 ) args[ len ] = arguments[ len + 1 ];
      if (shouldNotBeStubbed(el, stubs, modifiedComponents)) {
        return originalCreateElement.apply(void 0, [ el ].concat( args ))
      }

      if (isConstructor(el) || isComponentOptions(el)) {
        if (stubAllComponents) {
          var stub = createStubFromComponent(el, el.name || 'anonymous', _Vue);
          return originalCreateElement.apply(void 0, [ stub ].concat( args ))
        }
        var Constructor = shouldExtend(el, _Vue) ? extend(el, _Vue) : el;

        return originalCreateElement.apply(void 0, [ Constructor ].concat( args ))
      }

      if (typeof el === 'string') {
        var original = resolveComponent(el, originalComponents);

        if (!original) {
          return originalCreateElement.apply(void 0, [ el ].concat( args ))
        }

        if (isDynamicComponent(original)) {
          return originalCreateElement.apply(void 0, [ el ].concat( args ))
        }

        var stub$1 = createStubIfNeeded(stubAllComponents, original, _Vue, el);

        if (stub$1) {
          Object.assign(vm.$options.components, ( obj = {}, obj[el] = stub$1, obj));
          modifiedComponents.add(el);
        }
      }

      return originalCreateElement.apply(void 0, [ el ].concat( args ))
    }-->
`;

exports[`Loading size demo works fine 1`] = `
<div
  class="loading-demo"
>
  <span>
    小
  </span>
   
  <span
    class="t-loading t-size-s"
  >
    <svg
      class="t-icon t-icon-loading"
    >
      <use
        xlink:href="#t-icon-loading"
      />
    </svg>
  </span>
   
  <span>
    中
  </span>
   
  <span
    class="t-loading t-size-m"
  >
    <svg
      class="t-icon t-icon-loading"
    >
      <use
        xlink:href="#t-icon-loading"
      />
    </svg>
  </span>
   
  <span>
    大
  </span>
   
  <span
    class="t-loading t-size-l"
  >
    <svg
      class="t-icon t-icon-loading"
    >
      <use
        xlink:href="#t-icon-loading"
      />
    </svg>
  </span>
</div>
`;

exports[`Loading text demo works fine 1`] = `
<span
  class="t-loading t-size-m t-loading-text"
>
  <svg
    class="t-icon t-icon-loading"
  >
    <use
      xlink:href="#t-icon-loading"
    />
  </svg>
  <span>
    加载中
  </span>
</span>
`;

exports[`Loading wrap demo works fine 1`] = `
<div
  class="loading-demo"
>
  <div
    class="t-loading-parent__relative"
  >
    <div>
      this is loading component
    </div>
    <div
      class="t-loading__wrapper t-loading-mask"
    >
      <span
        class="t-loading t-size-m"
      >
        <svg
          class="t-icon t-icon-loading"
        >
          <use
            xlink:href="#t-icon-loading"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
